app/helpers/sessions_helper.rb

```diff
@@ -38,7 +39,30 @@ def clear_cart
     end
     
     def log_out
+        user = User.find_by(id: session[:user_id])
+        user.update_attribute(:cart, hto_s(session[:cart]))
```

Comment:
@modsoussi That makes sense about the cart never being hashed. You might also try looking at `to_json` or `as_json`:
```ruby
# Returns a String   "{\"make\":\"bmw\"}" 
{:make => 'bmw'}.to_json

# Returns a Hash   {"make"=>"bmw"} 
{:make => 'bmw'}.as_json
```